src/main/scala/com/qubole/spark/hiveacid/reader/v2/OrcColumnarBatchReader.java

```diff
@@ -0,0 +1,779 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.qubole.spark.hiveacid.reader.v2;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import com.qubole.shaded.orc.OrcConf;
+import com.qubole.shaded.orc.OrcFile;
+import com.qubole.shaded.orc.Reader;
+import com.qubole.shaded.orc.TypeDescription;
+import com.qubole.shaded.orc.mapred.OrcInputFormat;
+import com.qubole.shaded.hadoop.hive.common.type.HiveDecimal;
+import com.qubole.shaded.hadoop.hive.ql.exec.vector.*;
+import com.qubole.shaded.hadoop.hive.serde2.io.HiveDecimalWritable;
+import com.qubole.shaded.hadoop.hive.ql.io.orc.OrcSplit;
+import com.qubole.shaded.hadoop.hive.ql.io.orc.VectorizedOrcAcidRowBatchReader;
+import com.qubole.shaded.hadoop.hive.ql.plan.MapWork;
+import com.qubole.shaded.hadoop.hive.ql.exec.Utilities;
+import com.qubole.shaded.hadoop.hive.ql.exec.vector.VectorizedRowBatchCtx;
+import com.qubole.shaded.orc.OrcProto;
+import org.apache.hadoop.mapred.Reporter;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.io.NullWritable;
+import com.qubole.shaded.orc.OrcUtils;
+
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.execution.vectorized.ColumnVectorUtils;
+import org.apache.spark.sql.execution.vectorized.OffHeapColumnVector;
+import org.apache.spark.sql.execution.vectorized.OnHeapColumnVector;
+import org.apache.spark.sql.execution.vectorized.WritableColumnVector;
+import org.apache.spark.sql.types.*;
+import org.apache.spark.sql.vectorized.ColumnarBatch;
+import com.qubole.shaded.hadoop.hive.ql.io.sarg.*;
+
+/**
+ * After creating, `initialize` and `initBatch` should be called sequentially. This internally uses
+ * the Hive ACID Vectorized ORC reader to support reading of deleted and updated data.
+ */
+public class OrcColumnarBatchReader extends RecordReader<Void, ColumnarBatch> {
```

Comment:
This looks similar to org.apache.spark.sql.execution.datasources.orc.OrcColumnarBatchReader ? Should we use that instead . what are the extra changes here ?

Reply:
No it has some code from OrcColumnarBatchReader , but it has logic to support sargs and ACID table reader.

Reply:
The search arguments are set in reader option, so its better to set with the reader code.